hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -197,19 +204,28 @@ public HoodieActiveTimeline() {
   private void readObject(java.io.ObjectInputStream in) throws IOException, ClassNotFoundException {
     in.defaultReadObject();
   }
+  
+  private boolean shouldIncludeCompletionTime(HoodieInstant instant) {
+    return instant.isCompleted()
+        && metaClient.getTimelineLayoutVersion().getVersion() == TimelineLayoutVersion.VERSION_2;
+  }
```

Comment:
Do we need this decision

```java
metaClient.getTimelineLayoutVersion().getVersion() == TimelineLayoutVersion.VERSION_2
```

The old timeline does not even have `VERSION_2` in the code path.